### PR TITLE
Initial Jenkinsfile

### PR DESCRIPTION
**Pull request by DarthHater**
Start automating the creation of this image.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    }
+    if (branch != 'master') {
+      return
+    }
+    input 'Push image and tags?'
+    stage('Push image') {
+      def dockerhubApiToken
+      withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: 'docker-hub-credentials',
+          usernameVariable: 'DOCKERHUB_API_USERNAME', passwordVariable: 'DOCKERHUB_API_PASSWORD']]) {
+        OsTools.runSafe(this, "docker tag ${imageId} ${organization}/${dockerHubRepository}:${version}")
```
Comment (whyjustin) [inline Jenkinsfile]: This is probably a bug with our script but this version is incorrect if we are going an update.

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+        response = httpRequest customHeaders: [[name: 'authorization', value: "JWT ${dockerhubApiToken}"]],
+            acceptType: 'APPLICATION_JSON', contentType: 'APPLICATION_JSON', httpMode: 'PATCH',
+            requestBody: "{ \"full_description\": \"${readme}\" }",
+            url: "https://hub.docker.com/v2/repositories/${organization}/${dockerHubRepository}/"
+      }
+    }
+    stage('Push tags') {
+      withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: credentialsId,
+                        usernameVariable: 'GITHUB_API_USERNAME', passwordVariable: 'GITHUB_API_PASSWORD']]) {
+        OsTools.runSafe(this, "git tag ${version}")
```
Comment (whyjustin) [inline Jenkinsfile]: Same with the versions here

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+  error 'Could not determine version.'
+}
+
+def updateRepositoryManagerVersion(dockerFileLocation) {
+  def dockerFile = readFile(file: dockerFileLocation)
+
+  def metaVersionRegex = /(version=")(\d\.\d{1,3}\.\d\-\d{2})(" \\)/
+  def metaShortVersionRegex = /(release=")(\d\.\d{1,3}\.\d)(" \\)/
+
+  def versionRegex = /(ARG NEXUS_VERSION=)(\d\.\d{1,3}\.\d\-\d{2})/
```
Comment (whyjustin) [inline Jenkinsfile]: I believe this is the only updated needed (metaVersion, metaShortVersion, sha) are not used

File: Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+ * "Sonatype" is a trademark of Sonatype, Inc.
+ */
+@Library('zion-pipeline-library')
+import com.sonatype.jenkins.pipeline.GitHub
+import com.sonatype.jenkins.pipeline.OsTools
+
+properties([
+  parameters([
+    string(defaultValue: '', description: 'New Nexus Repository Manager Version', name: 'nexus_repository_manager_version'),
+    string(defaultValue: '', description: 'New Nexus Repository Manager Version Sha256', name: 'nexus_repository_manager_version_sha')
```
Comment (whyjustin) [inline Jenkinsfile]: NXRM 2.x Docker doesn't verify download by sha

File: Jenkinsfile (hunk 2 above)
Comment (DarthHater) [reply Jenkinsfile]: In: 15e5e3c

File: Jenkinsfile (hunk 1 above)
Comment (DarthHater) [reply Jenkinsfile]: In: 15e5e3c

**Comment by DarthHater** [comment]
Closing in favor of #71

**Pull request closed without merging**